src/hydra-queue-runner/dispatcher.cc

```diff
@@ -282,6 +282,19 @@ system_time State::doDispatch()
             if (keepGoing) break;
         }
 
+        for (auto & stepInfo : runnableSorted) {
+            auto & step(stepInfo.step);
+            bool couldRunStep = false;
+            for (auto & mi : machinesSorted)
+                if (mi.machine->supportsStep(step)) {
+                    couldRunStep = true;
+                    break;
+                }
+            if (couldRunStep) continue;
```

Comment:
Wouldn’t it make it more readable by putting the print inside `if` (Line 290). That way, you can get rid of the variable all together (because that `if` does a break).

Reply:
In this case I am printing on the empty set, not on membership, so the variable is needed to confirm that condition before printing.  The first member only confirms it is non-empty and shortcuts additional membership tests.